mbuild/compound.py

```diff
@@ -403,7 +404,8 @@ def generate_bonds(self, name_a, name_b, dmin, dmax):
                 bond_tuple = (p1, p2) if id(p1) < id(p2) else (p2, p1)
                 if bond_tuple in added_bonds:
                     continue
-                if (p2.name == name_b) and (dmin <= self.min_periodic_distance(p2.pos, p1.pos) <= dmax):
+                if (p2.name == name_b) and (dmin <= self.min_periodic_distance(p2.pos,
```

Comment:
Let's just break this into two lines or store the value of the distance evaluation to an intermediate.